src/meshes/t8code_mesh.jl

```diff
@@ -0,0 +1,383 @@
+                                                prod(trees_per_dimension))
+
+  # Get cell length in reference mesh: Omega_ref = [-1,1]^2.
+  dx = 2 / trees_per_dimension[1]
+  dy = 2 / trees_per_dimension[2]
+
+  num_local_trees = t8_cmesh_get_num_local_trees(cmesh)
+
+  # Non-periodic boundaries.
+  boundary_names = fill(Symbol("---"), 2 * NDIMS, prod(trees_per_dimension))
+
+  for itree = 1:num_local_trees
+    veptr = t8_cmesh_get_tree_vertices(cmesh, itree-1)
+    verts = unsafe_wrap(Array,veptr,(3,1 << NDIMS))
+
+    # Calculate node coordinates of reference mesh.
+    cell_x_offset = (verts[1,1] - 1/2*(trees_per_dimension[1]-1)) * dx
+    cell_y_offset = (verts[2,1] - 1/2*(trees_per_dimension[2]-1)) * dy
+
+    for j in eachindex(nodes), i in eachindex(nodes)
+      tree_node_coordinates[:, i, j, itree] .= mapping(cell_x_offset + dx * nodes[i]/2,
+                                                       cell_y_offset + dy * nodes[j]/2)
+    end
+
+    if !periodicity[1]
+      boundary_names[1, itree] = :x_neg
+      boundary_names[2, itree] = :x_pos
+    end
+
+    if !periodicity[2]
+      boundary_names[3, itree] = :y_neg
+      boundary_names[4, itree] = :y_pos
+    end
+  end
+
+  return T8codeMesh{NDIMS}(cmesh, scheme, forest, tree_node_coordinates, nodes, boundary_names, "", unsaved_changes)
+
+end
+
+"""
+    T8codeMesh{NDIMS}(cmesh::Ptr{t8_cmesh},
+                     mapping=nothing, polydeg=1, RealT=Float64,
+                     initial_refinement_level=0, unsaved_changes=true)
+Main mesh constructor for the `T8codeMesh` that imports an unstructured, conforming
+mesh from an Abaqus mesh file (`.inp`). Each element of the conforming mesh parsed
+from the `meshfile` is created as a [`p4est`](https://github.com/cburstedde/p4est)
+tree datatype.
+Note that the `mapping` and `polydeg` keyword arguments are only used by the `p4est_mesh_from_standard_abaqus`
+function. The `p4est_mesh_from_hohqmesh_abaqus` function obtains the mesh `polydeg` directly from the `meshfile`
+and constructs the transfinite mapping internally.
+The particular strategy is selected according to the header present in the `meshfile` where
+the constructor checks whether or not the `meshfile` was created with
+[HOHQMesh.jl](https://github.com/trixi-framework/HOHQMesh.jl).
+If the Abaqus file header is not present in the `meshfile` then the `P4estMesh` is created
+with the function `p4est_mesh_from_standard_abaqus`.
+The default keyword argument `initial_refinement_level=0` corresponds to a forest
+where the number of trees is the same as the number of elements in the original `meshfile`.
+Increasing the `initial_refinement_level` allows one to uniformly refine the base mesh given
+in the `meshfile` to create a forest with more trees before the simulation begins.
+For example, if a two-dimensional base mesh contains 25 elements then setting
+`initial_refinement_level=1` creates an initial forest of `2^2 * 25 = 100` trees.
+# Arguments
```

Comment:
```suggestion
`initial_refinement_level=1` creates an initial forest of `2^2 * 25 = 100` trees.

# Arguments
```
Please use a little bit more whitespace also in docstrings to make them easier to read as code.